docs/security.md

```diff
@@ -0,0 +1,194 @@
+# Security and Protection Features
+
+Many targets support some way of disabling JTAG/SWD access or protecting the flash from read-back or write.
+
+If you attempt to run pyOCD against a locked target, you'll encounter a message like this:
```

Comment:
`auto_unlock` defaults to True for every command except commander. So for most users that run the gdbserver, pyOCD will be able to connect by automatically mass erasing if the target is locked.